mongodbatlas/resource_mongodbatlas_online_archive.go

```diff
@@ -0,0 +1,414 @@
+package mongodbatlas
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/mwielbut/pointy"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	errorOnlineArchivesCreate = "error creating MongoDB Online Archive: %s"
```

Comment:
Small nit: We almost always want to include Atlas in the error like this that has MongoDB: 	
errorOnlineArchivesCreate = "error creating MongoDB **Atlas** Online Archive: %s"